kstars: 3.4.3 -> 3.5.0

hjones2199 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Add new kstars required dependency (stellarsolver) to nixpkgs, and update kstars to 3.5.0. Also updates kstars to use newer version of qt (5.15) and cleans up some mistakes I made in previous update to kstars (deviating from nixpkgs wiki recommendation to use imported mkDerivation instead of wrapQtApps hook).

ksutils.cpp patch also updated to support xplanet binary in the nix store.

Also adding myself as a maintainer to receive pings about kstars, I can force push that out if it's inappropriate.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

hjones2199 committed:
stellarsolver: init at 1.5

SuperSandro2000 commented on pkgs/applications/science/astronomy/kstars/default.nix:
```diff
@@ -1,30 +1,30 @@
   src = fetchurl {
-    url = "https://mirrors.mit.edu/kde/stable/kstars/kstars-${version}.tar.xz";
-    sha256 = "0j5yxg6ay6sic194skz6vjzg6yvrpb3gvypvs0frjrcjbsl1j4f8";
+    url = "https://mirrors.ocf.berkeley.edu/kde/stable/kstars/kstars-${version}.tar.xz";
```
```suggestion
    url = "mirror//kde/stable/kstars/kstars-${version}.tar.xz";
```

This should automatically choose an appropriate mirror.

hjones2199 replied:
Awesome, (assuming you meant mirror://) wasnt aware of that capability. Its now correctly pointing at mirror://kde/stable/kstars/kstars-${version}.tar.xz

SuperSandro2000 commented on pkgs/development/libraries/stellarsolver/default.nix:
```diff
@@ -0,0 +1,37 @@
+  ];
+
+  cmakeFlags = [
+    "-DBUILD_TESTER=off"
```
Why did you disable this?

hjones2199 replied:
Its off by default in CMakeLists.txt upstream, my latest commit makes it an option (withTester) that can be passed to the derivation, defaulting to off. I tested it both on and off and it checks out.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 104848` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>kstars</li>
    <li>stellarsolver</li>
  </ul>
</details>

SuperSandro2000 commented:
Result of `nixpkgs-review pr 104848` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>kstars</li>
    <li>stellarsolver</li>
  </ul>
</details>

doronbehar commented:
@hjones2199 thanks for working on this. Some of your changes were already done, and there's merge conflicts now. If you could please fix them.

hjones2199 committed:
kstars: 3.4.3 -> 3.5.0

hjones2199 commented:
> @hjones2199 thanks for working on this. Some of your changes were already done, and there's merge conflicts now. If you could please fix them.

Thanks for the heads up, rebased onto master and fixed conflicts

doronbehar commented on pkgs/applications/science/astronomy/kstars/default.nix:
```diff
@@ -9,20 +9,20 @@
 
   patches = [
-    ./indi-fix.patch
+    ./fs-fixes.patch
```
This patch should be explained:

- Who wrote the patch?
- Should we expect the next release to not need the patch?
- Is upstream aware of the changes required here for NixOS?

If upstream was submitted the patch, you should use `fetchpatch`.

hjones2199 replied:
I added a comment to the file in the most recent push explaining its purpose. To answer your questions about its nature/origins:

- I wrote the patch specifically for nix based installations
- Future releases will likely need the patch (though it may need to be reorganized if ksutils.cpp changes)
- I spoke with upstream in their matrix channel about the problem, and they are open to upstreaming a more general solution if me or someone else comes along to fix it.

The nature of the problem is that kstars runs other applications (like indi server for telescope control) as subprocesses, with their locations defined in ksutils.cpp. As it stands now, it uses a single $prefix/bin directory to find all of these binaries, whereas Nix uses different nix store paths for each of these dependencies. The patch lets the nix derivation pass individual nix store paths to cmake like "-DINDI_NIX_ROOT=${indilib}" for each of these programs. 

An alternative route I didnt go would be to have an optional FHS wrapper for kstars that throws them all in the same prefix, but i feel that the patch is a more simple solution for the time being. I'm also trying to work out a better solution upstream, at which point the patch can simply be removed.

doronbehar replied:
> I'm also trying to work out a better solution upstream, at which point the patch can simply be removed.

Fair enough.

> An alternative route I didnt go with would be to have an optional buildEnv wrapper for kstars that throws them all in the same prefix, but i feel that the patch is a more simple solution for the time being.

I agree.

hjones2199 commented on pkgs/development/libraries/stellarsolver/default.nix:
```diff
@@ -0,0 +1,37 @@
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/rlancaste/stellarsolver";
+    description = "Astrometric plate solving library";
+    license = licenses.gpl3;
```
Fixed, stellarsolver license is now licenses.gpl3Plus
